src/goats_tom/tasks/check_version.py

```diff
@@ -0,0 +1,112 @@
+"""
+Periodic task to check for the latest GOATS version and update the cache.
+"""
+
+__all__ = ["check_version"]
+
+import logging
+import uuid
+
+import dramatiq
+from django.core.cache import caches
+
+from goats_common import VersionChecker
+from goats_scheduler.scheduling import cron
+from goats_tom.realtime import NotificationInstance
+
+logger = logging.getLogger(__name__)
+
+CACHE_KEY = "version_info"
+LOCK_KEY = "lock:version_info"
+
+
+@cron(second="*/30")  # Every 30 seconds for testing.
+@dramatiq.actor
+def check_version() -> None:
+    """
+    Periodic task to check for the latest GOATS version and update the cache.
+    """
+
+    version_checker = VersionChecker()
+    result = version_checker.as_dict()
+
+    if result["status"] != "success":
+        # Do not update cache if there was an error checking versions.
+        if result.get("errors") is not None:
+            errors_str = "; ".join(result.get("errors", []))
+            logger.warning("Version check failed: %s", errors_str)
+        else:
+            logger.warning("Version check failed with unknown error.")
+        return
+
+    # Store the result in the cache.
+    _update_cache(result)
+
+
+def _update_cache(result: dict[str, str | None | bool | list[str]]) -> None:
+    """
+    Update the version info cache in Redis with locking.
+
+    Parameters
+    ----------
+    result : dict[str, str | None | bool | list[str]]
+        The version information to store in the cache.
+    """
+    cache = caches["redis"]
+    owner_id = str(uuid.uuid4())
+
+    # Create a lock.
+    if not cache.add(LOCK_KEY, owner_id, timeout=60):
```

Comment:
I added an owner ID to the lock value just for a simple ownership check before we delete the lock. I worry about a potential race condition.